RealmSwift/Sync.swift

```diff
@@ -283,12 +299,14 @@ public struct SyncConfiguration {
         case .pinCertificate(let path):
             certificate = path
         }
-        return RLMSyncConfiguration(user: user, realmURL: realmURL,
-                                    isPartial: !fullSynchronization,
-                                    urlPrefix: urlPrefix,
-                                    stopPolicy: stopPolicy,
-                                    enableSSLValidation: validateSSL,
-                                    certificatePath: certificate)
+        let c = RLMSyncConfiguration(user: user, realmURL: realmURL,
```

Comment:
[nit] We should add overload the current `[RLMSyncConfiguration initWith ...]` with another constructor that takes the `cancelAsyncOpenOnNonFatalErrors` param.

Reply:
This initializer isn't part of the public API so it wouldn't need to be a new overload. I don't really see a point to passing it through the initializer; it'd just be strictly more code with no benefits that I see.

Reply:
Fair point, will resolve